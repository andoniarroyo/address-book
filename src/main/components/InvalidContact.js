// @flow
// external modules
import React from 'react';
import { Link } from 'react-router-dom';
// styles
import '../styles/components/invalidContact.scss';

/**
 * The invalid contact render method
 * @param {ContactPropsType} props The properties to render the object
 * @returns {React$Element<any>} HTML markup for the component
 */
const InvalidContact = () => (
  <article className="invalid-container">
    <header>
        Ops! This is not a valid contact...
    </header>
    <section>
          You can create a <Link to="/add"> new contact</Link> or you can go to
          the <Link to="/"> home page</Link>
    </section>
  </article>
);

export default InvalidContact;
