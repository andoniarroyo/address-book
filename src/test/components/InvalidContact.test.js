// external modules
import React from 'react';
import renderer from 'react-test-renderer';
// internal modules
import InvalidContact from '../../main/components/InvalidContact';
import staticRouterWrapper from '../utils/staticRouterWrapper';

describe('<InvalidContact />', () => {
  it('renders the component', () => {
    const tree = renderer.create(staticRouterWrapper(<InvalidContact />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
