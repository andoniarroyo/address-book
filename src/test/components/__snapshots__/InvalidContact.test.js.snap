// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvalidContact /> renders the component 1`] = `
<article
  className="invalid-container"
>
  <header>
    Ops! This is not a valid contact...
  </header>
  <section>
    You can create a 
    <a
      href="/add"
      onClick={[Function]}
    >
       new contact
    </a>
     or you can go to the 
    <a
      href="/"
      onClick={[Function]}
    >
       home page
    </a>
  </section>
</article>
`;
